spec/services/events/create_service_spec.rb

```diff
@@ -6,309 +6,53 @@
-        expect(result).to be_success
+  let(:metadata) { {} }
 
-        event = result.event
-
-        aggregate_failures do
-          expect(event.customer_id).to eq(customer.id)
-          expect(event.organization_id).to eq(organization.id)
-          expect(event.code).to eq(billable_metric.code)
-          expect(event.subscription_id).to eq(subscription.id)
-          expect(event.timestamp).to be_a(Time)
-          expect(result.event.external_subscription_id).to eq(subscription.external_id)
-          expect(result.event.external_customer_id).to eq(customer.external_id)
-        end
-      end
-    end
-
-    context 'when creating an event to an active subscription with the same external id' do
-      let(:subscription) do
-        create(:subscription, customer:, organization:, status: :terminated, started_at: 1.month.ago)
-      end
-
-      let(:active_subscription) do
-        create(
-          :active_subscription,
-          customer:,
-          organization:,
-          started_at: 1.week.ago,
-          external_id: subscription.external_id,
-        )
-      end
-
-      let(:create_args) do
-        {
-          code: billable_metric.code,
-          transaction_id: SecureRandom.uuid,
-          external_subscription_id: subscription.external_id,
-          properties: { foo: 'bar' },
-          timestamp: 1.day.ago.to_i,
-        }
-      end
+  describe '#call' do
+    it 'creates an event' do
+      result = nil
 
-      before { active_subscription }
+      aggregate_failures do
+        expect do
+          result = create_service.call(params: create_args, timestamp: creation_timestamp, metadata:)
+        end.to change(Event, :count).by(1)
 
-      it 'creates an event to the active subscription' do
-        result = create_service.call(params: create_args, timestamp:, metadata: {})
         expect(result).to be_success
-
-        event = result.event
-
-        aggregate_failures do
-          expect(event.customer_id).to eq(customer.id)
-          expect(event.organization_id).to eq(organization.id)
-          expect(event.code).to eq(billable_metric.code)
-          expect(event.subscription_id).to eq(active_subscription.id)
-          expect(event.timestamp).to be_a(Time)
-          expect(result.event.external_subscription_id).to eq(active_subscription.external_id)
-          expect(result.event.external_customer_id).to eq(customer.external_id)
-        end
-      end
-    end
-
-    context 'when customer has only one active subscription and subscription is not given' do
-      it 'creates a new event and assigns subscription' do
-        result = create_service.call(
-          params: create_args,
-          timestamp:,
-          metadata: {},
+        expect(result.event).to have_attributes(
+          external_customer_id:,
+          external_subscription_id:,
+          transaction_id:,
+          code:,
+          timestamp: Time.at(timestamp),
```

Comment:
Rails/TimeZone: Do not use `Time.at` without zone. Use one of `Time.zone.at`, `Time.current`, `Time.at.in_time_zone`, `Time.at.utc`, `Time.at.getlocal`, `Time.at.xmlschema`, `Time.at.iso8601`, `Time.at.jisx0301`, `Time.at.rfc3339`, `Time.at.httpdate`, `Time.at.to_i`, `Time.at.to_f` instead. (https://rails.rubystyle.guide#time, http://danilenko.org/2012/7/6/rails_timezones)